drivers/telescope/telescope_simulator.cpp

```diff
@@ -366,49 +366,65 @@ bool ScopeSim::ReadScopeStatus()
         case SCOPE_SLEWING:
         case SCOPE_PARKING:
             /* slewing - nail it when both within one pulse @ SLEWRATE */
-            nlocked = 0;
+            nlocked = 0;        // seems to be some sort of state variable
 
-            dx = targetRA - currentRA;
+            deltaRa = targetRA - currentRA;
 
             // Always take the shortcut, don't go all around the globe
             // If the difference between target and current is more than 12 hours, then we need to take the shortest path
-            if (dx > 12)
-                dx -= 24;
-            else if (dx < -12)
-                dx += 24;
+            if (deltaRa > 12)
+                deltaRa -= 24;
+            else if (deltaRa < -12)
+                deltaRa += 24;
 
-            // In meridian flip, alway force eastward motion (increasing RA) until target is reached.
+            // In meridian flip, move to the position by doing a full rotation
             if (forceMeridianFlip)
             {
-                dx = fabs(dx);
-                if (dx == 0)
+                // set deltaRa according to the target pier side so that the
+                // slew is away from the meridian until the direction to go is towards
+                // the target.
+                switch (currentPierSide)
                 {
-                    // first step eastward if we start the meridian flip
-                    // ensure that the first step is bigger than the standard step to ensure flipping
-                    dx    = dt+0.5;
```

Comment:
This must not get lost with the change, since we need to ensure that with the first step fabs(deltaRa) * 15. > da_ra. If that's not the case, the simulator will not send the simulator in a 360° loop westward but will immediately jump to the target location without showing the behaviour of a meridian flip.

Test with 250ms polling period in the INDI Options tab to observe this behaviour.